drivers/timer/riscv_machine_timer.c

```diff
@@ -138,14 +84,6 @@ static void set_mtimecmp(uint64_t time)
 #endif
 }
 
-static void set_divider(void)
-{
-#ifdef MTIMER_HAS_DIVIDER
-	*(volatile uint32_t *)MTIMEDIV_REG =
-		CONFIG_RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER;
-#endif
```

Comment:
Is this being removed because it's dead code?  If so, in a separate commit from the DTS changes please.

Reply:
Yes, as far as I can tell it is dead code,  and there should be a more generic implementation of the ability to set the divider via clock drivers (clock_set_rate #72102 ) put in this driver later (I am aware that for at least the platforms i work on, there is a settable multiplier that sets the mtime rate from main clock). will do.

Reply:
Done.